abdulfatai360/banka
# #165187789 Integrate CI and test coverage/reporting tools

#### What does this PR do?
Integrate Continuous Integration and test coverage and test reporting tools

#### Description of Task to be completed?
- Have TravisCI integrated
- Integrate (Istanbul)[https://istanbul.js.org/docs/tutorials/mocha/] with Mocha for test coverage reporting
- Have (CodeClimate)[https://codeclimate.com] integrated for checking code quality
- Install (Coveralls)[https://coveralls.io) for adding coverage reporting to project's continuous-integration flow

#### How should this be manually tested?
On Terminal:
- Run `git clone https://github.com/abdulfatai360/banka.git` to clone this repo onto your local machine
- Navigate to the repo directory by running `cd banka/`
- Ensure your working directory is clean by running  `git status`  command
- Get a copy of the PR by typing  `git fetch origin pull/31/head:ch-integrate-ci-test-reporting-165187789`
- Now that you have a copy of the branch, switch to it using  `git checkout ch-integrate-ci-test-reporting-165187789`
- Your directory will now be an exact copy of the PR
- Run `npm install` to install all project's dependencies onto your local machine
- Type `npm run build` to run the build script and populate the `dist/` directory containing transpiled code
- Run `npm test` command to start automated test and get test coverage report on the terminal

#### Any background context you want to provide?
Some files were refactored to improve the project's code quality percentage

#### What are the relevant pivotal tracker stories?
#165187789
